job_simulation.rb

```diff
@@ -0,0 +1,78 @@
+require './stack.rb'
+require './Queue.rb'
+
+class Employees
+
+  attr_accessor :total_employees, :available_positions  #if company wants to adjust number of available positions
+  attr_reader :workers, :waiting
+
+  def initialize(company_info)
+    @available_positions = company_info[:available_positions].to_i
+    @total_employees = company_info[:total_employees].to_i
+    start_process
+  end
+
+  def start_process
+    @waiting = Queue.new # enqueue, dequeue
+    @workers = Stack.new # push, pop
+    @total_employees_array = (1..@total_employees).to_a
+    employee_initial_position
+  end
+
+  def employee_initial_position
+
+    @total_employees_array.each do |emp|
+      if emp <= @available_positions
+        @workers.push(emp)
+      else
+        @waiting.enqueue(emp)
+      end
+    end
+
+  end
```

Comment:
This works, but why create the initial array at all? Why not just put everyone directly into the queue and then pull out the first 6 for the stack after that?